### ISSUE TITLE
logging seems to be broken on master

### ISSUE THREAD

**Issue by dpsutton** [metabase/metabase#50721]
### Describe the bug

There's no logging:

Trying on both 

```shell
❯ java -version
openjdk version "11.0.25" 2024-10-15
OpenJDK Runtime Environment Temurin-11.0.25+9 (build 11.0.25+9)
OpenJDK 64-Bit Server VM Temurin-11.0.25+9 (build 11.0.25+9, mixed mode)
```
and

```shell
❯ java -version
openjdk version "21.0.2" 2024-01-16 LTS
OpenJDK Runtime Environment Temurin-21.0.2+13 (build 21.0.2+13-LTS)
OpenJDK 64-Bit Server VM Temurin-21.0.2+13 (build 21.0.2+13-LTS, mixed mode)
```

Here's the output of running a jar from master as of today:

(note: this is a fresh install)
```shell
❯ MB_JETTY_PORT=3006 java -jar metabase.jar
Warning: environ value /Users/dan/.sdkman/candidates/java/current for key :java-home has been overwritten with /Users/dan/.sdkman/candidates/java/21.0.2-tem
2024-12-02T02:21:03.533582Z main ERROR Unrecognized format specifier [d]
2024-12-02T02:21:03.534516Z main ERROR Unrecognized conversion specifier [d] starting at position 16 in conversion pattern.
2024-12-02T02:21:03.535144Z main ERROR Unrecognized format specifier [thread]
2024-12-02T02:21:03.535242Z main ERROR Unrecognized conversion specifier [thread] starting at position 25 in conversion pattern.
2024-12-02T02:21:03.535347Z main ERROR Unrecognized format specifier [level]
2024-12-02T02:21:03.535437Z main ERROR Unrecognized conversion specifier [level] starting at position 35 in conversion pattern.
2024-12-02T02:21:03.535540Z main ERROR Unrecognized format specifier [logger]
2024-12-02T02:21:03.535625Z main ERROR Unrecognized conversion specifier [logger] starting at position 47 in conversion pattern.
2024-12-02T02:21:03.535713Z main ERROR Unrecognized format specifier [msg]
2024-12-02T02:21:03.535798Z main ERROR Unrecognized conversion specifier [msg] starting at position 54 in conversion pattern.
2024-12-02T02:21:03.535886Z main ERROR Unrecognized format specifier [n]
2024-12-02T02:21:03.535969Z main ERROR Unrecognized conversion specifier [n] starting at position 56 in conversion pattern.
2024-12-02T02:21:03.545052Z main ERROR Reconfiguration failed: No configuration found for 'Default' at 'null' in 'null'
WARNING: infinite? already refers to: #'clojure.core/infinite? in namespace: kixi.stats.core, being replaced by: #'kixi.stats.math/infinite?

UPDATE SUMMARY
Run:                        436
Previously run:               0
Filtered out:                52
-------------------------------
Total change sets:          488


FILTERED CHANGE SETS SUMMARY
DBMS mismatch:               52

```

### To Reproduce

grab a jar and `MB_JETTY_PORT=3006 java -jar metabase.jar`

### Expected behavior

lots of logs

### Logs

there are no logs. See above.

### Information about your Metabase installation

```JSON
latest master, from https://github.com/metabase/metabase/actions/runs/12098926495
```

### Severity

p1

**Comment by dpsutton** [comment]
It should be noted that on master when running from source you will see good logs:

```
2024-12-02 02:27:28,372 INFO jdbcjobstore.JobStoreTX :: ClusterManager: ......Cleaned-up 2 other failed job(s).
2024-12-02 02:27:28,373 INFO core.QuartzScheduler :: Scheduler MetabaseScheduler_$_mbp-m1.lan1733106446855 started.
2024-12-02 02:27:28,373 INFO metabase.task :: Task scheduler started
2024-12-02 02:27:28,374 INFO jdbcjobstore.JobStoreTX :: Handling 9 trigger(s) that missed their scheduled fire-time.
2024-12-02 02:27:28,400 INFO task.sweep-query-analysis :: Calculating analysis for cards without any
2024-12-02 02:27:28,412 INFO task.sweep-query-analysis :: Recalculating potentially stale analysis
2024-12-02 02:27:28,447 INFO task.refresh-slack-channel-user-cache :: Slack is not configured, not refreshing slack user/channel cache.
2024-12-02 02:27:28,528 INFO task.sync-databases :: Not scheduling tasks for audit database
2024-12-02 02:27:28,548 INFO metabase.core :: Metabase Initialization COMPLETE in 51.5 s
2024-12-02 02:27:28,559 INFO metabase.events :: Loading events namespace: metabase.events.alerts-deleted-on-card-save 👂
2024-12-02 02:27:28,592 INFO metabase.events :: Loading events namespace: metabase.events.audit-log 👂
"malli: dev-mode started\n"
dev=> 2024-12-02 02:27:28,713 INFO metabase.events :: Loading events namespace: metabase.events.driver-notifications 👂
2024-12-02 02:27:28,727 INFO metabase.events :: Loading events namespace: metabase.events.last-login 👂
2024-12-02 02:27:28,737 INFO metabase.events :: Loading events namespace: metabase.events.notification 👂
2024-12-02 02:27:28,737 INFO metabase.events :: Loading events namespace: metabase.events.persisted-info 👂
2024-12-02 02:27:28,751 INFO metabase.events :: Loading events namespace: metabase.events.recent-views 👂
2024-12-02 02:27:28,779 INFO metabase.events :: Loading events namespace: metabase.events.revision 👂
2024-12-02 02:27:28,809 INFO metabase.events :: Loading events namespace: metabase.events.schema 👂
2024-12-02 02:27:28,810 INFO metabase.events :: Loading events namespace: metabase.events.slack 👂
2024-12-02 02:27:28,817 INFO metabase.events :: Loading events namespace: metabase.events.sync-database 👂
```

I suspect that this might be related to the build changes to switch over from depstar:

Artifacts from that PR are found here: https://github.com/metabase/metabase/actions/runs/12053514467
and from before that PR are here: https://github.com/metabase/metabase/actions/runs/12051136822

After the change we have
```
❯ MB_JETTY_PORT=3006 java -jar metabase.jar
2024-12-02T02:31:09.863769Z main ERROR Unrecognized format specifier [d]
2024-12-02T02:31:09.864992Z main ERROR Unrecognized conversion specifier [d] starting at position 16 in conversion pattern.
2024-12-02T02:31:09.865733Z main ERROR Unrecognized format specifier [thread]
2024-12-02T02:31:09.865866Z main ERROR Unrecognized conversion specifier [thread] starting at position 25 in conversion pattern.
2024-12-02T02:31:09.865974Z main ERROR Unrecognized format specifier [level]
2024-12-02T02:31:09.866067Z main ERROR Unrecognized conversion specifier [level] starting at position 35 in conversion pattern.
2024-12-02T02:31:09.866181Z main ERROR Unrecognized format specifier [logger]
2024-12-02T02:31:09.866286Z main ERROR Unrecognized conversion specifier [logger] starting at position 47 in conversion pattern.
2024-12-02T02:31:09.866388Z main ERROR Unrecognized format specifier [msg]
2024-12-02T02:31:09.866498Z main ERROR Unrecognized conversion specifier [msg] starting at position 54 in conversion pattern.
2024-12-02T02:31:09.866605Z main ERROR Unrecognized format specifier [n]
2024-12-02T02:31:09.866709Z main ERROR Unrecognized conversion specifier [n] starting at position 56 in conversion pattern.
2024-12-02T02:31:09.877554Z main ERROR Reconfiguration failed: No configuration found for 'Default' at 'null' in 'null'
WARNING: infinite? already refers to: #'kixi.stats.math/infinite? in namespace: kixi.stats.core, being replaced by: #'clojure.core/infinite?
```

and before that change we have

```
 MB_JETTY_PORT=3007 java -jar metabase.jar
Warning: environ value /Users/dan/.sdkman/candidates/java/current for key :java-home has been overwritten with /Users/dan/.sdkman/candidates/java/21.0.2-tem
2024-12-01 20:31:53,560 INFO metabase.util :: Maximum memory available to JVM: 8.0 GB
2024-12-01 20:31:54,741 INFO util.encryption :: Saved credentials encryption is DISABLED for this Metabase instance. 🔓
 For more information, see https://metabase.com/docs/latest/operations-guide/encrypting-database-details-at-rest.html
2024-12-01 20:31:55,182 WARN db.env :: WARNING: Using Metabase with an H2 application database is not recommended for production deployments. For production deployments, we highly recommend using Postgres, MySQL, or MariaDB instead. If you decide to continue to use H2, please be sure to back up the database file regularly. For more information, see https://metabase.com/docs/latest/operations-guide/migrating-from-h2.html
WARNING: infinite? already refers to: #'kixi.stats.math/infinite? in namespace: kixi.stats.core, being replaced by: #'clojure.core/infinite?
2024-12-01 20:31:57,745 INFO driver.impl :: Registered abstract driver :sql  🚚
2024-12-01 20:31:57,767 INFO driver.impl :: Registered abstract driver :sql-jdbc (parents: [:sql]) 🚚
2024-12-01 20:31:57,770 INFO metabase.util :: Load driver :sql-jdbc took 7.4 ms
2024-12-01 20:31:57,770 INFO driver.impl :: Registered driver :h2 (parents: [:sql-jdbc]) 🚚
2024-12-01 20:31:57,895 INFO driver.impl :: Registered driver :mysql (parents: [:sql-jdbc]) 🚚
2024-12-01 20:31:57,914 INFO driver.impl :: Registered driver :postgres (parents: [:sql-jdbc]) 🚚
```

**Comment by dpsutton** [comment]
Based on https://stackoverflow.com/questions/48033792/log4j2-error-statuslogger-unrecognized-conversion-specifier I think depstar was merged this stuff for us for free in the past.

In the "after" jar the `META-INF/org/apache/logging/log4j/core/config/plugins/Log4j2Plugins.dat` is 3247 bytes. In the "before" it is `28463` bytes long.

See the workaround: https://github.com/seancorfield/depstar/blob/develop/src/hf/depstar/uberjar.clj#L151-L166

I think we can use https://github.com/seancorfield/build-uber-log4j2-handler to help us get there.
